From 9e2b1ea55e1676649f4130bd15c2cfcceb7bfed0 Mon Sep 17 00:00:00 2001
From: Lyon Wang <lyon.wang@freescale.com>
Date: Mon, 15 Dec 2014 16:52:07 +0800
Subject: [PATCH 9/9] handle audio/video decoder error

When there is input data and no output data to the end of the stream, it will
send GST_ELEMENT_ERROR, So the clips playing will quit.
However, if only one of the tracks is corrupt, there is no need to quit other
tracks playing.

The patch comments the GST_ELEMENT_ERROR() and just add GST_ERROR_OBJECT()
information instead.

https://bugzilla.gnome.org/show_bug.cgi?id=741542

Upstream-Status: Pending

Signed-off-by: Lyon Wang <lyon.wang@freescale.com>
---
 gst-libs/gst/audio/gstaudiodecoder.c | 5 +++--
 gst-libs/gst/video/gstvideodecoder.c | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)
 mode change 100644 => 100755 gst-libs/gst/audio/gstaudiodecoder.c
 mode change 100644 => 100755 gst-libs/gst/video/gstvideodecoder.c

diff --git a/gst-libs/gst/audio/gstaudiodecoder.c b/gst-libs/gst/audio/gstaudiodecoder.c
old mode 100644
new mode 100755
index b68f9eb..506c13d
--- a/gst-libs/gst/audio/gstaudiodecoder.c
+++ b/gst-libs/gst/audio/gstaudiodecoder.c
@@ -2304,9 +2304,10 @@ gst_audio_decoder_sink_eventfunc (GstAudioDecoder * dec, GstEvent * event)
       GST_AUDIO_DECODER_STREAM_UNLOCK (dec);
 
       if (dec->priv->ctx.had_input_data && !dec->priv->ctx.had_output_data) {
-        GST_ELEMENT_ERROR (dec, STREAM, DECODE,
+        /* GST_ELEMENT_ERROR (dec, STREAM, DECODE,
             ("No valid frames decoded before end of stream"),
-            ("no valid frames found"));
+            ("no valid frames found")); */
+        GST_ERROR_OBJECT(dec, "No valid frames decoded before end of stream");
       }
 
       /* send taglist if no valid frame is decoded util EOS */
diff --git a/gst-libs/gst/video/gstvideodecoder.c b/gst-libs/gst/video/gstvideodecoder.c
old mode 100644
new mode 100755
index 514c06f..fa29083
--- a/gst-libs/gst/video/gstvideodecoder.c
+++ b/gst-libs/gst/video/gstvideodecoder.c
@@ -1251,9 +1251,10 @@ gst_video_decoder_sink_event_default (GstVideoDecoder * decoder,
 
       /* Error out even if EOS was ok when we had input, but no output */
       if (ret && priv->had_input_data && !priv->had_output_data) {
-        GST_ELEMENT_ERROR (decoder, STREAM, DECODE,
+        /* GST_ELEMENT_ERROR (decoder, STREAM, DECODE,
             ("No valid frames decoded before end of stream"),
-            ("no valid frames found"));
+            ("no valid frames found")); */
+        GST_ERROR_OBJECT(decoder, "No valid frames decoded before end of stream");
       }
 
       /* Forward EOS immediately. This is required because no
-- 
1.9.1

